model-optimizer/mo/utils/cli_parser.py

```diff
@@ -845,6 +845,23 @@ def get_freeze_placeholder_values(argv_input: str, argv_freeze_placeholder_with_
     return placeholder_values, input_node_names
 
 
+def parse_dimension(dim: str):
+    if '..' in dim:
+        numbers_reg = r'[0-9]+'
+        dims = dim.split('..')
+        min_val = np.int64(dims[0]) if re.match(numbers_reg, dims[0]) else np.int64(0)
+        max_val = np.int64(dims[1]) if re.match(numbers_reg, dims[1]) else np.iinfo(np.int64).max
+        assert min_val >= 0, "Incorrect min value of the dimension {}".format(dim)
+
+        if min_val == np.int64(0) and max_val == np.iinfo(np.int64).max:
+            return np.int64(-1)
+
```

Comment:
please check that `min_val <= max_val`.
In case they are equal, should we return just one value?

Reply:
Done.